api/osrd_infra/views/study.py

```diff
@@ -0,0 +1,206 @@
+
+from osrd_infra.models import Project, Scenario, Study, Timetable
+from osrd_infra.schemas.study import StudyState, StudyType
+from osrd_infra.serializers import (
+    ProjectSerializer,
+    ScenarioSerializer,
+    StudySerializer,
+)
+from osrd_infra.views.pagination import CustomPageNumberPagination
+
+
+class ScenarioView(
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.ListModelMixin,
+    GenericViewSet,
+):
+    serializer_class = ScenarioSerializer
+    filter_backends = [filters.OrderingFilter]
+    ordering_fields = ["name", "creation_date", "last_modification"]
+    pagination_class = CustomPageNumberPagination
+
+    def get_queryset(self):
+        """this function will disappear to be replaced by the generic search (#2901)"""
+        project = get_object_or_404(Project.objects.all(), pk=self.kwargs["project_pk"])
+        study = get_object_or_404(Study.objects.all(), pk=self.kwargs["study_pk"], project=project)
+        queryset = Scenario.objects.filter(study__project=project, study=study)
+        name = self.request.query_params.get("name", None)
+        description = self.request.query_params.get("description", None)
+        tags = self.request.query_params.get("tags", None)
+        filters = Q()
+        if name:
+            filters = filters | Q(name__icontains=name)
+        if description:
+            filters = filters | Q(description__icontains=description)
+        if tags:
+            filters = filters | Q(tags__icontains=tags)
+        return queryset.filter(filters)
+
+    def create(self, request, *args, **kwargs):
+        project = get_object_or_404(Project.objects.all(), pk=self.kwargs["project_pk"])
+        study = get_object_or_404(Study.objects.all(), pk=self.kwargs["study_pk"], project=project)
+        scenario_name = request.data["name"]
+        infra_id = request.data.pop("infra")
+        timetable_name = f"timetable for {scenario_name}"
+        timetable = Timetable.objects.create(name=timetable_name)
+        scenario = Scenario.objects.create(timetable=timetable, study_id=study.id, infra_id=infra_id, **request.data)
+        serializer = ScenarioSerializer(scenario, context={"request": request})
+        return Response(serializer.data)
+
+    def retrieve(self, request, pk, project_pk=None, study_pk=None):
+        scenario = get_object_or_404(self.get_queryset(), pk=pk)
+        serializer = ScenarioSerializer(scenario, context={"request": request})
+
+        train_schedules = [
+            {
+                "id": train.pk,
+                "train_name": train.train_name,
+                "departure_time": train.departure_time,
+                "train_path": train.path_id,
+            }
+            for train in scenario.timetable.train_schedules.all()
+        ]
+
+        return Response({**serializer.data, "infra_name": scenario.infra.name, "train_schedules": train_schedules})
```

Comment:
This is already in `serializer.data`
```suggestion
        return Response({**serializer.data, "train_schedules": train_schedules})
```

Reply:
Thanks :)